pdns/slavecommunicator.cc

```diff
@@ -939,22 +942,15 @@ void CommunicatorClass::slaveRefresh(PacketHandler *P)
         d_failedSlaveRefresh.erase(di.zone);
     }
 
-    bool hasSOA = false;
-    SOAData sd;
-    try{
-      hasSOA = B->getSOA(di.zone, sd);
-    }
-    catch(...) {}
-
     uint32_t theirserial = ssr.d_freshness[di.id].theirSerial;
-    uint32_t ourserial = sd.serial;
+    uint32_t ourserial = di.serial;
```

Comment:
I was a bit worried that this change was undoing https://github.com/PowerDNS/pdns/pull/7245, but it looks fine since di.serial is now properly updated above.

Reply:
Yes, I hope I understood the intention of #7245 correctly.